### PR TITLE
Update client side validator API

### LINKED ISSUE #685: Finish up client validators API to handle pages
Node sends the validators response in batches with cursor indicating pages. Client method should allow for requesting all pages

### PR DESCRIPTION
**Pull request by daria305**
Part of #685

- Move restapi leftovers to api

Corresponding iota.go [PR](https://github.com/iotaledger/iota.go/pull/682/files)

### CODE REVIEW

File: components/restapi/core/accounts.go [hunk 1]
```diff
@@ -62,10 +61,10 @@ func congestionByAccountAddress(c echo.Context) (*api.CongestionResponse, error)
 func validators(c echo.Context) (*api.ValidatorsResponse, error) {
 	var err error
 	pageSize := restapi.ParamsRestAPI.MaxPageSize
-	if len(c.QueryParam(restapipkg.QueryParameterPageSize)) > 0 {
-		pageSize, err = httpserver.ParseUint32QueryParam(c, restapipkg.QueryParameterPageSize)
+	if len(c.QueryParam(api.QueryParameterPageSize)) > 0 {
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
	if len(c.QueryParam(api.ParameterPageSize)) > 0 {
```

File: components/restapi/core/accounts.go [hunk 2]
```diff
@@ -62,10 +61,10 @@ func congestionByAccountAddress(c echo.Context) (*api.CongestionResponse, error)
...
+		pageSize, err = httpserver.ParseUint32QueryParam(c, api.QueryParameterPageSize)
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
		pageSize, err = httpserver.ParseUint32QueryParam(c, api.ParameterPageSize)
```

File: components/restapi/core/accounts.go [hunk 3]
```diff
@@ -62,10 +61,10 @@ func congestionByAccountAddress(c echo.Context) (*api.CongestionResponse, error)
...
 		if err != nil {
-			return nil, ierrors.Wrapf(err, "failed to parse page size %s", c.Param(restapipkg.QueryParameterPageSize))
+			return nil, ierrors.Wrapf(err, "failed to parse page size %s", c.Param(api.QueryParameterPageSize))
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
			return nil, ierrors.Wrapf(err, "failed to parse page size %s", c.Param(api.ParameterPageSize))
```

File: components/restapi/core/accounts.go [hunk 4]
```diff
@@ -62,10 +61,10 @@ func congestionByAccountAddress(c echo.Context) (*api.CongestionResponse, error)
 func validators(c echo.Context) (*api.ValidatorsResponse, error) {
 	var err error
 	pageSize := restapi.ParamsRestAPI.MaxPageSize
```
Comment (muXxer) [inline components/restapi/core/accounts.go]: Maybe we should add and use the function from the indexer.

```go

func pageSizeFromContext(c echo.Context) uint32 {
	maxPageSize := uint32(deps.RestAPILimitsMaxResults)
	if len(c.QueryParam(api.ParameterPageSize)) > 0 {
		pageSizeQueryParam, err := httpserver.ParseUint32QueryParam(c, api.ParameterPageSize, maxPageSize)
		if err != nil {
			return maxPageSize
		}

		if pageSizeQueryParam < maxPageSize {
			// use the smaller page size given by the request
			maxPageSize = pageSizeQueryParam
		}
	}

	return maxPageSize
}
```

File: components/restapi/core/accounts.go [hunk 5]
```diff
@@ -75,10 +74,10 @@ func validators(c echo.Context) (*api.ValidatorsResponse, error) {
 	// no cursor provided will be the first request
 	requestedSlot := latestCommittedSlot
 	var cursorIndex uint32
-	if len(c.QueryParam(restapipkg.QueryParameterCursor)) != 0 {
-		requestedSlot, cursorIndex, err = httpserver.ParseCursorQueryParam(c, restapipkg.QueryParameterCursor)
+	if len(c.QueryParam(api.QueryParameterCursor)) != 0 {
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
	if len(c.QueryParam(api.ParameterCursor)) != 0 {
```

File: components/restapi/core/accounts.go [hunk 6]
```diff
@@ -75,10 +74,10 @@ func validators(c echo.Context) (*api.ValidatorsResponse, error) {
...
+		requestedSlot, cursorIndex, err = httpserver.ParseCursorQueryParam(c, api.QueryParameterCursor)
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
		requestedSlot, cursorIndex, err = httpserver.ParseCursorQueryParam(c, api.ParameterCursor)
```

File: components/restapi/core/accounts.go [hunk 7]
```diff
@@ -75,10 +74,10 @@ func validators(c echo.Context) (*api.ValidatorsResponse, error) {
...
 		if err != nil {
-			return nil, ierrors.Wrapf(err, "failed to parse cursor %s", c.Param(restapipkg.QueryParameterCursor))
+			return nil, ierrors.Wrapf(err, "failed to parse cursor %s", c.Param(api.QueryParameterCursor))
```
Comment (muXxer) [inline components/restapi/core/accounts.go]:
```suggestion
			return nil, ierrors.Wrapf(err, "failed to parse cursor %s", c.Param(api.ParameterCursor))
```

File: components/restapi/core/accounts.go (hunk 4 above)
Comment (muXxer) [reply components/restapi/core/accounts.go]: Or we should even add this function to the httpserver package, so we can reuse it in the indexer as well.

**Review by muXxer — CHANGES_REQUESTED** [review]

File: components/restapi/core/accounts.go (hunk 4 above)
Comment (daria305) [reply components/restapi/core/accounts.go]: I opened PR in inx-app: https://github.com/iotaledger/inx-app/pull/115

**Review by muXxer — APPROVED** [review]

**Pull request merged**